crates/composefs/src/fsverity/mod.rs

```diff
@@ -172,6 +172,24 @@ mod tests {
         fd
     }
 
+    /// Wrap the real method inside of the tests and do a little
+    /// sleep/retry dance.  This is known to be race-prone due to the
+    /// way the tests are run in parallel across different runner
+    /// threads, see:
+    /// https://github.com/containers/composefs-rs/issues/106
+    fn enable_verity<H: FsVerityHashValue>(fd: impl AsFd) -> Result<(), EnableVerityError> {
+        for attempt in 1..=3 {
+            match super::enable_verity::<H>(&fd) {
+                Err(EnableVerityError::FileOpenedForWrite) if attempt != 3 => {
+                    std::thread::sleep(std::time::Duration::from_millis(1));
+                }
+                other => return other,
+            }
+        }
+
+        unreachable!("We always return on the last iteration above");
```

Comment:
We can avoid this `unreachable!` with an explicit `loop`, something like this

```
fn enable_verity<H: FsVerityHashValue>(fd: impl AsFd) -> Result<(), EnableVerityError> {
    let mut attempt = 1;
    loop {
        match super::enable_verity::<H>(&fd) {
            Err(EnableVerityError::FileOpenedForWrite) if attempt < 3 => {
                std::thread::sleep(std::time::Duration::from_millis(1));
                attempt += 1;
            }
            other => return other,
        }
    }
}
```

But anyways I think it wouldn't really be too much more code to do the copy-based fallback right? And make it generic outside of the tests